cime_config/machines/config_machines.xml

```diff
@@ -1269,23 +1269,25 @@
         <command name="load">cmake/3.21.3</command>
         <command name="load">cray-hdf5-parallel/1.12.2.1</command>
         <command name="load">cray-netcdf-hdf5parallel/4.9.0.1</command>
-	<command name="load">cray-parallel-netcdf/1.12.3.1</command>
+        <command name="load">cray-parallel-netcdf/1.12.3.1</command>
         <command name="unload">darshan-runtime</command>
       </modules>
     </module_system>
-    
+
     <RUNDIR>$CIME_OUTPUT_ROOT/$CASE/run</RUNDIR>
     <EXEROOT>$CIME_OUTPUT_ROOT/$CASE/bld</EXEROOT>
     <TEST_TPUT_TOLERANCE>0.1</TEST_TPUT_TOLERANCE>
     <MAX_GB_OLD_TEST_DATA>0</MAX_GB_OLD_TEST_DATA>
     <environment_variables>
       <env name="NETCDF_PATH">$ENV{NETCDF_DIR}</env>
       <env name="PNETCDF_PATH">$ENV{PNETCDF_DIR}</env>
+      <env name="HDF5_ROOT"/> <!-- frontier cmake cannot find_package(HDF5) for some reason, so disable it -->
       <env name="MPICH_GPU_SUPPORT_ENABLED">1</env>
       <env name="MPICH_VERSION_DISPLAY">1</env>
       <env name="MPICH_OFI_CXI_COUNTER_REPORT">2</env>
+      <env name="MPICH_CXX">$SHELL{which hipcc}</env>
       <env name="LD_LIBRARY_PATH">$ENV{CRAY_LD_LIBRARY_PATH}:$ENV{LD_LIBRARY_PATH}</env>
-
+      <env name="BLAS_LIBRARIES">$ENV{CRAY_LIBSCI_PREFIX_DIR}/lib/libsci_cray.a</env> <!-- CMake does not currently support Cray as a blas vendor -->
```

Comment:
CMake tends to be schizophrenic with LibSci, used to know about LibSci.

Anyway, you don't want to link a specific variant as compiler flags (with OpenMP, accelerator support etc.) dictate which BLAS lib gets linked. 

```
$ ls /opt/cray/pe/libsci/22.12.1.1/CRAY/9.0/x86_64/lib/
libsci_cray.a         libsci_cray_mpi_mp.so      libsci_cray_mpi.so.5    libsci_cray_mp.so.5.0  pkgconfig
libsci_cray_mp.a      libsci_cray_mpi_mp.so.5    libsci_cray_mpi.so.5.0  libsci_cray.so
libsci_cray_mpi.a     libsci_cray_mpi_mp.so.5.0  libsci_cray_mp.so       libsci_cray.so.5
libsci_cray_mpi_mp.a  libsci_cray_mpi.so         libsci_cray_mp.so.5     libsci_cray.so.5.0
```